From: Hariprasad S <hariprasad@chelsio.com>
Date: Mon, 23 Jun 2014 19:12:35 +0530
Subject: [PATCH 166/223] RDMA/cxgb4: Fix skb_leak in reject_cr()
Patch-mainline: v3.16-rc6
Git-commit: 233b43010330ed8cf39cf636880017df3e33f102
References: bsc#909577 FATE#317550

Based on origninal work by Steve Wise <swise@opengridcomputing.com>

Signed-off-by: Steve Wise <swise@opengridcomputing.com>
Signed-off-by: Hariprasad Shenai <hariprasad@chelsio.com>
Signed-off-by: Roland Dreier <roland@purestorage.com>
Acked-by: David Chang <dchang@suse.com>
Signed-off-by: John Jolly <jjolly@suse.de>
---
 drivers/infiniband/hw/cxgb4/cm.c |    1 -
 1 file changed, 1 deletion(-)

--- a/drivers/infiniband/hw/cxgb4/cm.c
+++ b/drivers/infiniband/hw/cxgb4/cm.c
@@ -1910,7 +1910,6 @@ static void reject_cr(struct c4iw_dev *d
 	     peer_ip);
 	BUG_ON(skb_cloned(skb));
 	skb_trim(skb, sizeof(struct cpl_tid_release));
-	skb_get(skb);
 	release_tid(&dev->rdev, hwtid, skb);
 	return;
 }
